packages/flutter/lib/src/widgets/bottom_navigation_bar_item.dart

```diff
@@ -36,6 +38,16 @@ class BottomNavigationBarItem {
   /// [IconTheme] size and color.
   final Widget icon;
 
+  /// An alternative icon displayed when this item is not selected in a material
+  /// fixed bottom navigation bar.
+  ///
+  /// To ensure that bottom navigation items without a title are accessible,
+  /// use a filled version of an icon as [icon] and a stroked version as
+  /// [inactiveIcon].
+  ///
+  /// Defaults to [icon].
+  final Widget inactiveIcon;
```

Comment:
(this comment is not actionable)
Interesting, I would have done it the other way around: icon specifies the icon that you'll see most of the time (tab is not selected) and then you have a special selectedIcon property for when the tab is selected. Obviously, your choice works as well and I and I have little arguments for one choice or the other...

Reply:
I originally chose this way because I was worried about backwards compatibility, but thinking about it more as long as I default to `icon` when `activeIcon` is not provided it should be fine.